sharded_queue/drivers/fifottl.lua

```diff
@@ -281,10 +281,16 @@ function method.take(args)
         next_event = dead_event
     end
 
-    task = box.space[args.tube_name]:update(task[index.task_id], {
+    return box.space[args.tube_name]:update(task[index.task_id], {
         { '=', index.status,     state.TAKEN },
         { '=', index.next_event, next_event  }
     })
+end
+
+function method.take(args)
+
+    local task = box.atomic(take, args)
+    if task == nil then return end
```

Comment:
The tests should be updated with new stats behavior. Also, I'm not sure about skipping wc_signal here.